src/wallet/wallet.cpp

```diff
@@ -1507,6 +1507,41 @@ bool DummySignInput(const SigningProvider& provider, CTxIn &tx_in, const CTxOut
     return true;
 }
 
+void FillInputToWeight(CTxIn& txin, int64_t target_weight)
+{
+    int64_t txin_weight = GetTransactionInputWeight(txin);
+
+    // Do nothing if the weight that should be added is less than the weight that already exists
+    if (target_weight <= txin_weight) {
```

Comment:
nit: let's assert that no witness data exists yet?

Reply:
Done